### PR TITLE
Introduce new script that validates the signatures match between the .pyi file and the bridge

### PR DESCRIPTION
**Pull request by jleibs**
### What
- Resolves: https://github.com/rerun-io/rerun/issues/7645
- Fixes the corresponding issues.

Detected several issues:
```
➜  pixi run check-py-signatures
✨ Pixi task (check-py-signatures in py): python scripts/ci/check_python_signatures.py
IndexColumnSelector.__new__(…) signature mismatch:
    Stub: (cls, index)
    Runtime: (*args, **kwargs)
ComponentColumnSelector.__new__(…) signature mismatch:
    Stub: (cls, entity_path, component_type)
    Runtime: (*args, **kwargs)
Recording.view(…) signature mismatch:
    Stub: (self, /, index, contents)
    Runtime: (self, /, *, index, contents)
load_recording(…) signature mismatch:
  Stub: (filename)
  Runtime: (path_to_rrd)
load_archive(…) signature mismatch:
  Stub: (filename)
  Runtime: (path_to_rrd)
```

Example CI failure:
image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7646?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7646?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/7646)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: pixi.toml [hunk 1]
```diff
@@ -305,6 +305,8 @@ py-build = "pixi run -e py py-build-common"
 # Dedicated alias for building the python bindings in release mode for the `py` environment.
 py-build-release = "pixi run -e py py-build-common-release"
 
+check-py-signatures = "python scripts/ci/check_python_signatures.py"
```
Comment (teh-cmc) [inline pixi.toml:308]: To be consistent with the rest, and more easily findable through autocomplete/tab spam
```suggestion
py-check-signatures = "python scripts/ci/check_python_signatures.py"

**Review by teh-cmc — APPROVED** [review]

File: scripts/ci/check_python_signatures.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+    if result == 0:
+        print("All stub signatures match!")
+
+    return result
+
+
+def main() -> int:
+    # load the stub file
+    path_to_stub = Path(__file__).parent / ".." / ".." / "rerun_py" / "rerun_bindings" / "rerun_bindings.pyi"
```
Comment (abey79) [inline scripts/ci/check_python_signatures.py:167]:
```suggestion
    path_to_stub = Path(__file__).parent.parent.parent / "rerun_py" / "rerun_bindings" / "rerun_bindings.pyi"
```

File: scripts/ci/check_python_signatures.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+import ast
+import importlib
+import inspect
+import sys
+from inspect import Parameter, Signature
+from pathlib import Path
+from typing import Any
+
+import parso
```
Comment (abey79) [inline scripts/ci/check_python_signatures.py:18]: Shouldn't this be added to the `pixi.toml`?

**Review by abey79 — APPROVED** [review]
Extremely based

File: scripts/ci/check_python_signatures.py (hunk 3 above)
Comment (jleibs) [reply scripts/ci/check_python_signatures.py:18]: It was already in the environment, but good call. I'll make it explicit.

**Pull request merged**